File: frontend/src/api/useAutoApi/apiCase.ts
import request from '/@/utils/request';

const BaseApiPath = "/apiCase"
/**
 * 项目使用接口
 * @method getProjectList 获取项目列表
 * @method getMenuTest 获取后端动态路由菜单(test)
 */
export function useApiCaseApi() {
  return {
    getList: (data?: object) => {
      return request({
        url: `${BaseApiPath}/list`,
        method: 'POST',
        data,
      });
    },
    saveOrUpdate(data?: object) {
      return request({
        url: `${BaseApiPath}/saveOrUpdate`,
        method: 'POST',
        data
      })
    },
    runSuites(data?: object) {
      return request({
        url: `${BaseApiPath}/runTestCase`,
        method: 'POST',
        data
      })
    },
    debugSuites(data?: object) {
      return request({
        url: `${BaseApiPath}/debugTestCase`,
        method: 'POST',
        data
      })
    },
    deleted: (data?: object) => {
      return request({
        url: `${BaseApiPath}/deleted`,
        method: 'POST',
        data,
      });
    },
    getSuitesInfo: (data?: object) => {
      return request({
        url: `${BaseApiPath}/getSuiteInfo`,
        method: 'POST',
        data,
      });
    },
  };
}

File: backend/autotest/views/api/api_case.py
from flask import Blueprint, request

from autotest.services.api_services.api_case import ApiCaseService
from autotest.utils.api import partner_success

bp = Blueprint('api_case', __name__, url_prefix='/api/apiCase')


@bp.route('/list', methods=['POST'])
def suites_list():
    """
    套件列表
    :return:
    """
    data = ApiCaseService.list(**request.json)
    return partner_success(data=data)


@bp.route('/saveOrUpdate', methods=['POST'])
def save_or_update():
    """
    更新保存套件
    :return:
    """
    data = ApiCaseService.save_or_update(**request.json)
    return partner_success(data=data.id)


@bp.route('/runTestCase', methods=['POST'])
def run_testcase():
    """
    运行用例
    :return:
    """
    data = ApiCaseService.run_case(**request.json)
    return partner_success(data)


@bp.route('/debugTestCase', methods=['POST'])
def debug_testcase():
    """
    调试用例
    :return:
    """
    data = ApiCaseService.debug_case(**request.json)
    return partner_success(data=data)


@bp.route('/deleted', methods=['POST'])
def deleted():
    """
    删除套件
    :return:
    """
    parsed_data = request.json
    s_id = parsed_data.get('id', None)
    ApiCaseService.deleted(s_id)
    return partner_success()


@bp.route('/getSuiteInfo', methods=['POST'])
def suite_info():
    """
    套件信息
    :return:
    """
    data = ApiCaseService.get_case_info(**request.json)
    return partner_success(data=data)
